src/day6.cpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include <vector>
+#include <fstream>
+#include <algorithm>
+
+
+std::vector<std::vector<std::string>> readPuzzleInputFromFile(const std::string& fileName) {
+    std::ifstream inFile(fileName);
+    std::string line;
+    std::vector<std::string> answersPerGroup;
+    std::vector<std::vector<std::string>> answers;
+    while(getline(inFile, line)) {
+        if(line.empty()) {
+            answers.push_back(answersPerGroup);
+            answersPerGroup.clear();
+            continue;
+        }
+        answersPerGroup.push_back(line);
+    }
+    if (!answersPerGroup.empty()) answers.push_back(answersPerGroup);
+    return answers;
+}
+
+int numberOfAnswers_AnsweredByAnyGroupMember(const std::vector<std::string>& answers) {
+    if ( answers.size() == 1 ) return answers[0].size();
+    
+    std::string uniqueAnswers = answers[0];
+    for (size_t i = 1; i < answers.size(); i++) {
+        for (const char& answer : answers[i]) {
+            if (uniqueAnswers.find(answer) == std::string::npos) {
```

Comment:
Note that `find` on an `std::string` is O(n) operation, because it is a linear scan through the string. You could use a `std::unordered_set` to store the answers, and then searching for an answer would take approximately constant time (since it's implemented with a hashmap). **However**, since `uniqueAnswers` isn't typically that big (at most 26 characters), and because linear memory scan are razor fast, it may still very well be that your solution is faster. Could be interesting to profile!